syft/source/exclude.go

```diff
@@ -1,5 +1,69 @@
 package source
 
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/anchore/syft/syft/internal/fileresolver"
+	"github.com/bmatcuk/doublestar/v4"
+)
+
 type ExcludeConfig struct {
 	Paths []string
 }
+
+func GetDirectoryExclusionFunctions(root string, exclusions []string) ([]fileresolver.PathIndexVisitor, error) {
```

Comment:
This is returning an internal package. I don't think this function should be a part of this package, probably is better to be internal (or left alone)